liberum_core/src/swarm_runner/mod.rs

```diff
@@ -33,8 +35,9 @@ const DEFAULT_MULTIADDR_STR_IP4: &str = "/ip4/0.0.0.0/udp/0/quic-v1";
 
 /// The context of the swarm which holds all the data required to handle swarm events
 /// and messages to the swarm runner
-struct SwarmContext {
+pub struct SwarmContext {
```

Comment:
Why does it need to be public?

Reply:
I think I have made it public in the first commits to make it possible to use it inside an outer parser which would handle all the objects instead of turning them into structs, which turned out to be a bad idea, at least in the existing codebase
https://github.com/Liberum-Neto/liberum-neto/pull/56/commits/5ca536a7c25282f4ba3479c57bf88af41d69c6db